version: '3.8'

services:
  db:
    image: postgres:13
    volumes:
      - postgres_data:/var/lib/postgresql/data/
    environment:
      - POSTGRES_DB=ac_db
      - POSTGRES_USER=ac_user
      - POSTGRES_PASSWORD=ac_pass
    healthcheck:
      test: ["CMD-SHELL", "pg_isready -U ac_user -d ac_db"]
      interval: 5s
      timeout: 5s
      retries: 5

  pgadmin:
    image: dpage/pgadmin4
    restart: always
    environment:
      PGADMIN_DEFAULT_EMAIL: adel@gmail.com
      PGADMIN_DEFAULT_PASSWORD: 12054801
    ports:
      - "5050:80"
    depends_on:
      - db

  web:
    build: .
    command: >
      sh -c "while ! nc -z db 5432; do sleep 1; done &&
             python manage.py migrate &&
             python manage.py runserver 0.0.0.0:8000"
    volumes:
      - ./backend:/code
    ports:
      - "8000:8000"
    environment:
      - DEBUG=1
      - SECRET_KEY=django-insecure-w@p+fa&&omp^oa7&n68==+0@4^+lert43yw8v8yjhrxip821po
      - ALLOWED_HOSTS=localhost,127.0.0.1
    depends_on:
      db:
        condition: service_healthy

volumes:
  postgres_data:
